OpenRA.Mods.Common/Traits/Pluggable.cs

```diff
@@ -46,6 +58,13 @@ public Pluggable(ActorInitializer init, PluggableInfo info)
 			var plugInit = init.Contains<PlugsInit>() ? init.Get<PlugsInit, Dictionary<CVec, string>>() : new Dictionary<CVec, string>();
 			if (plugInit.ContainsKey(Info.Offset))
 				initialPlug = plugInit[Info.Offset];
+
+			if (info.Requirements.Count > 0)
+			{
+				plugTypesAvailability = new Dictionary<string, bool>();
+				foreach (var plug in info.Requirements)
+					plugTypesAvailability.Add(plug.Key, true);
```

Comment:
It would be better to use `plugTypesAvailability[plug.Key] = true` here to avoid a crash if someone defines duplicate requirements.